Chore: Pre-audit plumbing tasks

SatwikPrabhu opened the pull request:
Fix for https://github.com/botanix-labs/botanix/issues/895

arminsabouri commented:
@SatwikPrabhu can you provide the command you ran? Im suprised its only 1 change

SatwikPrabhu commented:
> @SatwikPrabhu can you provide the command you ran? Im suprised its only 1 change

I just ran `cargo clippy --fix"

SatwikPrabhu commented:
Re running with `cargo clippy --workspace --fix --allow-staged --allow-dirty`. Getting more changes. Apologies!

arminsabouri commented:
Thanks. I'll review when its out of draft

SatwikPrabhu committed:
chore: cargo audit and fix vulnerabilities

Ran cargo audit and update dependencies with known vulnerabilities

SatwikPrabhu commented:
There were 3 vulnerabilities found during cargo audit. I could fix one of them. The other two I could not as they might require forking external crates unless they update and resolve them. Please find attached details of the audit results. There are 11 other warnings as well. Should I address the warnings?
[audit_results.txt](https://github.com/user-attachments/files/18700668/audit_results.txt)

arminsabouri commented:
> e 11 other warnings as well. Should I address the warning

If its easy to fix then yes please.

arminsabouri commented:
@SatwikPrabhu looks like we have a couple CI failures including clippy checks

SatwikPrabhu commented:
> If its easy to fix then yes please.

Fixed a few easy ones. Now we have 2 vulnerabilities and 7 warnings.

arminsabouri commented:
> > If its easy to fix then yes please.
> 
> Fixed a few easy ones. Now we have 2 vulnerabilities and 7 warnings.

@SatwikPrabhu would it be easy to document this in a ticket?

arminsabouri commented:
@SatwikPrabhu I may ask that we wait to merge this until we merge #506

SatwikPrabhu commented:
> @SatwikPrabhu would it be easy to document this in a ticket?

Yeah it shouldnt be hard. I'll do that.

arminsabouri commented:
Crates to target: authority, network, primitives, rpc

SatwikPrabhu committed:
chore: Enable clippy checks in ci pipelines

SatwikPrabhu committed:
chore: clippy fixes

Ran clippy fix on the following reth crates: authority, network,
primitives, rpc

arminsabouri reviewed (changes requested):
Thanks for staying with this one. I understand it can be a bit tideous. A couple comments in order of importance:
* A couple of unit tests are broken
* Your extended commit messages include previous commit messages. Do you mind cleanings that up. Ideally the clippy change command you wrong is included in the extended commit message. For example a full commit message for this would look like
"chore: fix project wide clipp warnings
ran: `cargo clippy fix -- ...` "

* If we are making clippy changes in these other crates can we add them to the CI change. No is an acceptable answer here if this increases the scope. 
* More of a question, how do we know locally if we make clippy happy? Is there an easy command to run? Can we add it to the makefile

For future refrence The rust bitcoin ecosystem put me on to [this guide](https://cbea.ms/git-commit/) for writing commit messages.

rwlockbg commented:
@SatwikPrabhu Could you add to the Makefile the command you are using to run the `cargo clippy` so that we could add this to our pre-hooks, run it locally and also make sure the pipelines are aware of this.

Also fix the conflicts.

SatwikPrabhu commented:
> If we are making clippy changes in these other crates can we add them to the CI change. No is an acceptable answer here if this increases the scope.

We can add them, but yeah we'll have to do another round of cleaning up the errors. But yeah adding another 3 or 4 should be doable pretty soon.

> More of a question, how do we know locally if we make clippy happy? Is there an easy command to run? Can we add it to the makefile

As far as I know, the one I'm using currently should be the one. The problem here was that even after we fixed all the warnings shown by clippy locally, we get additional errors when running the same command on CI. Will investigate more on this.

arminsabouri commented:
@SatwikPrabhu I see we have a conflict on `bin/btc-server/src/rpc/btc_server.bin`
in generally I try to not to commit this file unless im making direct changes to the grpc interface. Might just need to re-generate it and push
